### DISCUSSION TITLE
Suggestions for improving the number of respondents a survey can handle at once?

### DISCUSSION THREAD

**Discussion by willking98** [surveydown-dev/surveydown#114] [Q&A]
### Description

I have recently piloted my survey with a group of colleagues with the aim of testing whether the survey infrastructure would be able to support 10 or more different individuals all loading and completing the survey at the same time. In short the survey crashed and wasn't able to load when 10 people all tried to complete it at once. 

I've identified three options that I think might improve the survey's ability to support many respondents at once but I wondered if you had any advice for which is worthwhile or whether there are any other options available. 
1. Invest in a paid shinyapps.io hosting service.
2. Reduce the complexity of the survey (limit show_if conditionals)
3. Use a different PostgreSQL database platform which is better able to cope with high traffic

**Comment by jhelvy** [comment]
Oof, thanks for letting us know. I'd be curious to know more about the settings you used for shinyapps.io. Was it crashing there? Even on the free tier, you can bump up the RAM etc to give it more compute power. Also, the free tier of Supabase I believe only supports 20 connections, so there's a chance that also caused some constraints. 

Looking at the last version of your survey I saw, it did seem like there were ways to reduce some complexities in how the survey was created. You have a lot of show_if conditions, which could probably be simplified in some way. But even so, the platform should be able to support lots of concurrent sessions regardless of how complex the survey is, so there's proably still work to be done on our end. 

One other thing that could be helpful for us is to ask if your team could try taking one of our simpler demo surveys all at once? Like, could you make a copy of one of the simple ones, like maybe the [show_if demo](https://surveydown.org/demos), then have them take it? If that one crashes with just 10 concurrent sessions then there's something more fundamental we need to update.

**Reply by jhelvy** [reply]
Can you try bumping up the settings to the largest available for the free tier on shinyapps.io? It should look something like this:

![Screen Shot 2024-10-08 at 12 16 08](https://github.com/user-attachments/assets/4ca6f269-4b33-46c3-a726-b54a5eb13f81)

**Reply by willking98** [reply]
> Can you try bumping up the settings to the largest available for the free tier on shinyapps.io? It should look something like this:
> 
> ![Screen Shot 2024-10-08 at 12 16 08](https://private-user-images.githubusercontent.com/9012994/374641518-4ca6f269-4b33-46c3-a726-b54a5eb13f81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0MDkzNzIsIm5iZiI6MTcyODQwOTA3MiwicGF0aCI6Ii85MDEyOTk0LzM3NDY0MTUxOC00Y2E2ZjI2OS00YjMzLTQ2YzMtYTcyNi1iNTRhNWViMTNmODEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDhUMTczNzUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTdjOGI3N2ZhNGY4M2RlNDMwYjJmM2ZlMDg0YzgwYTQwNmM2OTBiMDlhNDg1ZmEzZjYzNWY4MDUzMWNkN2NkYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.5xzzfJOmOhHebCuyhKQgx-Cmw4I2CaEBDaBP0iT1pIc)

For me, the largest available free tier is 1GB (see pic).
![image](https://github.com/user-attachments/assets/72c18680-e134-46cc-9805-182ea35e7b80)

I have increased the start count under Instance Settings though (which was set to 1) so I will see if this works. 

Cheers for all the information and advice. I will try out a very simple survey with many connections at once and also try the survey with the higher start count when I can rope some colleagues into trying it out again.

**Reply by jhelvy** [reply]
Another option is https://connect.posit.cloud/. This is a pretty new site that posit recently released. It may have more options / larger memory limits on the free tier. Again, I think there's probably a different way to set up your survey to be less memory intense, but we should do some benchmarking on our end.

**Comment by jhelvy** [comment]
@willking98 we just made another significant change to the database writing in v0.3.5. Now the database only updates on each page turn, which should significantly improve the speed for anything happening on a single page as there won't be any database operations happening. That said, a survey a lot of show if conditions still may require more RAM on the shinyapps.io server, but hopefully the database won't be a factor slowing things down.

**Comment by jhelvy** [comment]
Added a [performance](https://surveydown.org/performance) page to the website.
